### ISSUE TITLE
Duplicated subscription using hook useSubscription

### ISSUE THREAD

**Issue by ANavakahtsky** [apollographql/apollo-client#6348]
Hello there!

I'm trying to use hook useSubscription in my component as following:

`const onSubscriptionData = useCallback(
    (options: { subscriptionData: any }) => {
      const {
        subscriptionData: { data, loading, error },
      } = options;
      if (!loading && !error && !!data) {
        fieldArray.fields.push(mapPropertyAddress(data.addressAdded));
      }
    },
    [fieldArray.fields]
  );

  useSubscription(createSubscription, { shouldResubscribe: false, onSubscriptionData });`

Described below is true for both v2.6 client(using @apollo/react-hooks v3.1.5) and v3.0beta (using @apollo/client v3.0.0-beta.50)

**Intended outcome:**
I see single subscription message in web socket connection and get data once for each subscription trigger.
**Actual outcome:**
I see 2 subscription messages sent to the server via web socket connection, get [2 messages](http://prntscr.com/spbffv) for updates and get my onSubscriptionData handler called twice after update. I multiple-times-checked and useSubscription hook called only once.

**How to reproduce the issue:**
It's quite hard to give an example here as you need to set up both working client and server to establish web socket connection.

**Versions**
System:
    OS: Windows 10 10.0.17763
  Binaries:
    Node: 12.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.19.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: Version 83.0.4103.61 (Official Build) (64-bit)
  npmPackages:
    @apollo/react-common: ^3.1.4 => 3.1.4
    @apollo/react-hooks: ^3.1.5 => 3.1.5
    apollo-cache-inmemory: ^1.6.6 => 1.6.6
    apollo-client: ^2.6.10 => 2.6.10
    apollo-link: ^1.2.14 => 1.2.14
    apollo-link-http: ^1.5.17 => 1.5.17
    apollo-link-ws: ^1.0.20 => 1.0.20
    apollo-utilities: ^1.3.4 => 1.3.4

Has anybody faced the same issue?

**Comment by stopdropandrew** [comment]
Looks like the same as https://github.com/apollographql/apollo-client/issues/6037

**Comment by hwillson** [comment]
Closing as a duplicate of #6037. Thanks!
